collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/IntentGroupTest.java

```diff
@@ -80,17 +95,109 @@ public void appMissingErrorText_ShouldComeFromSpecialFormText() {
     }
 
     @Test
-    public void externalApp_ShouldPopulateFields() {
+    public void externalApp_ShouldPopulateStringFields() {
         Intent resultIntent = new Intent();
-        resultIntent.putExtra("sometext", "Here is a text value");
 
-        int randomInteger = (new Random()).nextInt(255);
-        resultIntent.putExtra("someinteger", randomInteger);
+        int randomInteger = new Random().nextInt(255);
+        double scale = Math.pow(10, 2);
+        double randomDecimal = Math.round(new Random().nextDouble() * scale) / scale;
+        String randomText = RandomString.make();
+
+        resultIntent.putExtra("questionInteger", randomInteger);
+        resultIntent.putExtra("questionDecimal", randomDecimal);
+        resultIntent.putExtra("questionText", randomText);
 
         intending(not(isInternal())).respondWith(new Instrumentation.ActivityResult(Activity.RESULT_OK, resultIntent));
         onView(withText("This is buttonText")).perform(click());
 
-        onView(withText("Here is a text value")).check(matches(isDisplayed()));
         onView(withText(Integer.toString(randomInteger))).check(matches(isDisplayed()));
+        onView(withText(String.valueOf(randomDecimal))).check(matches(isDisplayed()));
+        onView(withText(randomText)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void externalApp_ShouldPopulateImageField() throws IOException {
```

Comment:
Could this just be one test where the group contains every kind of data? That might be harder to read though I guess... Interested in what you think!

Reply:
The test will be long but yeah in this case where it's about populating multiple fields at the same time it's maybe better to have one test.